### PR TITLE
Suggested edits for language and flow

### PR DESCRIPTION
**Pull request by joshkauffman**

### CODE REVIEW

File: content/guides/authentication.md [hunk 1]
```diff
@@ -8,17 +8,29 @@ weight: 2
...
 
 ## Issuing a Short-Lived JWT
 
-You use the following command to Generate a short-lived JTW from your API key. Do not forget to replace the API key by your own!
+Once you have this API key, call the [`https://auth.dfuse.io/v1/auth/issue` endpoint](#post-v1-auth-issue) to generate a fresh Authentication Token using the following command. **Do not forget to replace the API key by your own!**
 
 {{< tabs "issuing-long-jwt">}}
 {{< tab title="Request" lang="shell" >}}
-curl https://auth.dfuse.io/v1/auth/issue -s --data-binary '{"api_key":"web_abcdef12345678900000000000"}'
+curl <https://auth.dfuse.io/v1/auth/issue> -s --data-binary '{"api_key":"web_abcdef12345678900000000000"}'
```
Comment (olivierbeaulieu) [inline content/guides/authentication.md]: Why are you adding the `<` and `>`? I'm not sure why that would be necessary.

File: content/guides/authentication.md [hunk 2]
```diff
@@ -8,17 +8,29 @@ weight: 2
...
+First assign the API key a name so that you'll be able to differentiate it from other API keys that you generate.
+Then, select the key-type between `mobile`, `web`, or `server`.
+  * `Mobile` - adaptive rate limiting to ensure abuse prevention. The rate limiting thresholds will eventually be controllable by yourself. `Mobile` can handle a large number of IPs, with more granular rate limiting by IP.
+  * `Web` - adaptive rate limiting to ensure abuse prevention. The rate limiting thresholds will eventually be controllable by yourself. JWTs generated by a `Web` key will be scoped to a small number of IPs, with heavy limiting of calls per IP. `Web` keys will have their `Origin header` verified to help with abuse prevention.
+  * `Server` - Should be used for server-to-server data access. Aggressive rate limiting for JWT issuance, as the generated tokens should be cached and used for their lifespan. Accepts a large number of calls per IP.
+  
+For ease of identification, all keys will be prefixed by their key type. `Mobile` and `web` keys are safe to publish within your mobile or web application.
 
 ## Issuing a Short-Lived JWT
```
Comment (olivierbeaulieu) [inline content/guides/authentication.md]: Should we simplify the titles to "Issuing an API Key" and "Issuing a JWT"?

Also, I'm not sure issuing is the right word. The user doesn't issue the key - we do. The user `obtains` or `gets` a key.

File: content/guides/authentication.md [hunk 3]
```diff
@@ -29,39 +41,27 @@ curl https://auth.dfuse.io/v1/auth/issue -s --data-binary '{"api_key":"web_abcde
...
+
 ## Lifecycle of Short-Lived JWTs
 
-Each JWT has an expiration date, so it is important to take that into account before making a request to the _dfuse_ APIs. Here is the recommended procedure to take before making a request:
+Each JWT has an expiration date, so it is important to take that into account before making a request to the dfuse APIs. Here is the recommended procedure to take before making a request:
 
 1. Retrieve the JWT, and examine it's expiration time
-1. If the JWT is past it's expiration time, is near expiration or is absent from cache, fetch a new one through the `/v1/auth/issue` endpoint, and cache the response.
-1. Call _dfuse_ with the valid JWT token.
+1. If the JWT is past its expiration time, is near expiration, or is absent from cache, fetch a new one through the `/v1/auth/issue` endpoint, and cache the response.
```
Comment (olivierbeaulieu) [inline content/guides/authentication.md]: Should we make that link to the endpoint's page in reference?

File: content/guides/authentication.md [hunk 4]
```diff
@@ -29,39 +41,27 @@ curl https://auth.dfuse.io/v1/auth/issue -s --data-binary '{"api_key":"web_abcde
...
+1. Call dfuse with the valid JWT token.
```
Comment (olivierbeaulieu) [inline content/guides/authentication.md]: `Call the dfuse API`?

Or call dfuse, 1-800-555-9948? :)

File: content/guides/authentication.md [hunk 5]
```diff
@@ -73,15 +73,15 @@ eosc -H "Authorization: Bearer YOURTOKENHERE" -u <https://mainnet.eos.dfuse.io>
 {{< /tab >}}
 {{< /tabs >}}
 
-Don't forget to include your token in the above command with a valid JWT, retrieved using [`/v1/auth/issue`](#post-v1-auth-issue)
+Don't forget to replace the token in the above command with a valid JWT, retrieved using [`/v1/auth/issue`](#post-v1-auth-issue)
```
Comment (olivierbeaulieu) [inline content/guides/authentication.md]: Dead link, we should update it.

With Hugo, the way to do this is by using the `ref` shortcode. Example for here:

```
...retrieved using [`/v1/auth/issue`]({{< ref "/reference/eosio/rest/auth-issue-intro" >}})
```

`auth-issue-intro` being the name of the .md file you are trying to link to. Using this method will also add validation at build time that the link is not dead, protecting us against broken links when we move files, rename stuff, etc.

The format for all hugo shortcodes is `{{< shortcodeName "someoptions" >}}`

File: content/guides/authentication.md [hunk 6]
```diff
@@ -73,15 +73,15 @@ eosc -H "Authorization: Bearer YOURTOKENHERE" -u <https://mainnet.eos.dfuse.io>
...
 
 ## WebSocket Authentication
 
 This example uses the <https://github.com/hashrocket/ws> command-line WebSocket program. To authenticate with query string, use this code:
 
 {{< tabs "websocket-auth" >}}
 {{< tab lang="shell" >}}
-ws -o https://b2b.dfuse.io wss://mainnet.eos.dfuse.io/v1/stream?token=YOURTOKENHERE
+ws -o <https://b2b.dfuse.io> wss://mainnet.eos.dfuse.io/v1/stream?token=YOURTOKENHERE
```
Comment (olivierbeaulieu) [inline content/guides/authentication.md]: Same question as above

File: content/guides/authentication.md (hunk 2 above)
Comment (joshkauffman) [reply content/guides/authentication.md]: Agreed. I would use "Obtain an API Key" and "Obtain a JWT"

File: content/guides/authentication.md (hunk 1 above)
Comment (joshkauffman) [reply content/guides/authentication.md]: Removed. Git blame says it was you, but I won't tell anyone ;)

File: content/guides/authentication.md (hunk 3 above)
Comment (joshkauffman) [reply content/guides/authentication.md]: Still needs to be figured out and fixed.

**Pull request merged**